src/concurrency/concurrency_handler.rs

```diff
@@ -0,0 +1,51 @@
+use std::rc::Rc;
+
+use super::data_race;
+use crate::concurrency::GenmcCtx;
+use crate::{VisitProvenance, VisitWith};
+
+pub enum ConcurrencyHandler {
```

Comment:
This doesn't handle concurrency, it just handles data race detection and scheduling. So I think `DataRaceHandler` or so might be a better name?